### ISSUE TITLE
(beta) Build Error: Type 'typeof VError' is not a constructor function type.

### ISSUE THREAD

**Issue by EinfachHans** [sequelize/umzug#434]
- Umzug Version: **3.0.0-beta.14**
- Node Version: `v14.15.4`

Just started using umzug and get this error when i try to build:

```shell
node_modules/umzug/lib/umzug.d.ts:11:45 - error TS2507: Type 'typeof VError' is not a constructor function type.

11 export declare class MigrationError extends VError {
```

**Comment by kasradzenika** [comment]
same here

- umzug 3.0.0-beta.14
- node 14.15.4
- typescript 4.1.3

switching to umzug@2.3.0 worked for me

**Comment by mmkal** [comment]
Thanks @EinfachHans - could you share your tsconfig?

**Comment by EinfachHans** [comment]
@mmkal of course:

```
{
  "compilerOptions": {
    "target": "es2017",
    "lib": ["es2017", "esnext.asynciterable"],
    "typeRoots": ["./node_modules/@types"],
    "allowSyntheticDefaultImports": true,
    "experimentalDecorators": true,
    "emitDecoratorMetadata": true,
    "forceConsistentCasingInFileNames": true,
    "moduleResolution": "node",
    "module": "commonjs",
    "pretty": true,
    "sourceMap": false,
    "outDir": "./build",
    "allowJs": true,
    "noEmit": false,
    "esModuleInterop": true
  },
  "include": ["./src/**/*"],
  "exclude": ["node_modules"]
}
```

**Comment by mmkal** [comment]
@EinfachHans haven't had a chance to look at this yet but does the problem go away if allowSyntheticDefaultImports and/or esModuleInterop are turned off? Will need to be fixed either way but would help narrow down the problem.

**Comment by mmkal** [comment]
@EinfachHans @kasradzenika fixed in [3.0.0-beta.15](https://github.com/sequelize/umzug/releases/tag/v3.0.0-beta.15) - let me know if there are still issues.

Note that there's a breaking change in there too, so if you're using a custom storage it will need to be updated slightly. If you're just using sequelize no changes should be needed.
